src/components/LibrariesList.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from "react";
+import { PanelGroup } from "react-bootstrap";
+import { Store, Reducer } from "redux";
+import { State } from "../reducers/index";
+import { LibrariesData, LibraryData } from "../interfaces";
+import { connect } from "react-redux";
+import ActionCreator from "../actions";
+import LibrariesListItem from "./LibrariesListItem";
+
+export interface LibrariesListStateProps {
+  libraries?: LibrariesData;
+}
+
+export interface LibrariesListOwnProps {
+  store?: Store<State>;
+  csrfToken?: string;
+}
+
+export interface LibrariesListDispatchProps {
+  fetchData: () => Promise<LibrariesData>;
+}
+
+export interface LibrariesListState {
+  activeKey: string;
+}
+
+export interface LibrariesListProps extends LibrariesListStateProps, LibrariesListOwnProps, LibrariesListDispatchProps {};
+
+export class LibrariesList extends React.Component<LibrariesListProps, LibrariesListState> {
+
+  constructor(props: LibrariesListProps) {
+    super(props);
+    this.state = { activeKey: "" };
+    this.isActive = this.isActive.bind(this);
+    this.select = this.select.bind(this);
+  }
+
+  isActive(library: LibraryData) {
+    return this.props.libraries.libraries.indexOf(library) === parseInt(this.state.activeKey);
+  }
+
+  select(idx: string) {
+    this.setState({ activeKey: idx});
+    this.props.fetchData();
```

Comment:
Why fetch all the libraries again here?

Reply:
So that, if you edited the stages, the color-coding immediately changes to reflect that.

Reply:
The problem with this is that it's fetching all libraries data every time you open and close a panel. It's unnecessary if an admin wants to just look at all the library data and not make any changes.

Also, I noticed there was a flash every time I opened or closed a panel and that it wasn't smooth like the examples in the react-bootstrap docs. This fetch to the API causes that flash and non-smooth transition.

Reply:
This seems to be the component tree:
```
  LibrariesList
    LibrariesListItem
      LibraryDetailContainer
        LibraryDetailPage
```

From `LibraryDetailPage` when you submit you then trigger a toggle/select function which triggers all the way up to this `select` function. It's a long way for the panel just to close after submitting data.

You can probably pass a callback from `ListDetailPage` with the value of the Library and Registry stages up to `LibrariesListItem`. There you can intercept the `this.props.select` call, get the two stage values, and render the panel header with the appropriate color. Then you would not have to fetch all libraries data again. It also seems you don't need all that information anyway because when `LibraryDetailPage` renders, it fetches its own set of data.

Reply:
Ah, I wondered why there was a flash!  I will give this a try.